src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c

```diff
@@ -570,6 +570,25 @@ static void xclmgmt_icap_get_data(struct xclmgmt_dev *lro, void *buf)
 	XOCL_PUT_XCLBIN_ID(lro);
 }
 
+static void xclmgmt_clock_get_data_impl(struct xclmgmt_dev *lro, void *buf)
+{
+	struct xcl_pr_region *hwicap = NULL;
+
+	hwicap = (struct xcl_pr_region *)buf;
+	hwicap->freq_0 = xocl_clock_get_data(lro, CLOCK_FREQ_0);
+	hwicap->freq_1 = xocl_clock_get_data(lro, CLOCK_FREQ_1);
+	hwicap->freq_2 = xocl_clock_get_data(lro, CLOCK_FREQ_2);
+	hwicap->freq_cntr_0 = xocl_clock_get_data(lro, FREQ_COUNTER_0);
+	hwicap->freq_cntr_1 = xocl_clock_get_data(lro, FREQ_COUNTER_1);
+	hwicap->freq_cntr_2 = xocl_clock_get_data(lro, FREQ_COUNTER_2);
+}
+
+static void xclmgmt_icap_get_data(struct xclmgmt_dev *lro, void *buf)
+{
+	return xocl_has_icap(lro) ? xclmgmt_icap_get_data_impl(lro, buf) :
```

Comment:
Please use -ENODEV and remove xocl_has_icap().

Reply:
The current API will return 0 instead of -ENODEV, should I change it?
XOCL_GET_XCLBIN_ID